src/linux/network.rs

```diff
@@ -60,6 +60,80 @@ impl Networks {
     }
 }
 
+fn refresh_networks_list_from_sysfs(
+    interfaces: &mut HashMap<String, NetworkData>,
+    sysfs_net: &Path,
+) {
+    if let Ok(dir) = std::fs::read_dir(sysfs_net) {
+        let mut data = vec![0; 30];
+
+        for stats in interfaces.values_mut() {
+            stats.updated = false;
+        }
+
+        for entry in dir.flatten() {
+            let parent = &entry.path().join("statistics");
+            let entry = match entry.file_name().into_string() {
+                Ok(entry) => entry,
+                Err(_) => continue,
+            };
+            let rx_bytes = read(parent, "rx_bytes", &mut data);
+            let tx_bytes = read(parent, "tx_bytes", &mut data);
+            let rx_packets = read(parent, "rx_packets", &mut data);
+            let tx_packets = read(parent, "tx_packets", &mut data);
+            let rx_errors = read(parent, "rx_errors", &mut data);
+            let tx_errors = read(parent, "tx_errors", &mut data);
+            // let rx_compressed = read(parent, "rx_compressed", &mut data);
+            // let tx_compressed = read(parent, "tx_compressed", &mut data);
+            match interfaces.entry(entry) {
+                hash_map::Entry::Occupied(mut e) => {
+                    let mut interface = e.get_mut();
+                    old_and_new!(interface, rx_bytes, old_rx_bytes);
+                    old_and_new!(interface, tx_bytes, old_tx_bytes);
+                    old_and_new!(interface, rx_packets, old_rx_packets);
+                    old_and_new!(interface, tx_packets, old_tx_packets);
+                    old_and_new!(interface, rx_errors, old_rx_errors);
+                    old_and_new!(interface, tx_errors, old_tx_errors);
+                    // old_and_new!(e, rx_compressed, old_rx_compressed);
+                    // old_and_new!(e, tx_compressed, old_tx_compressed);
+                    interface.updated = true;
+                }
+                hash_map::Entry::Vacant(e) => {
+                    e.insert(NetworkData {
+                        rx_bytes,
+                        old_rx_bytes: rx_bytes,
+                        tx_bytes,
+                        old_tx_bytes: tx_bytes,
+                        rx_packets,
+                        old_rx_packets: rx_packets,
+                        tx_packets,
+                        old_tx_packets: tx_packets,
+                        rx_errors,
+                        old_rx_errors: rx_errors,
+                        tx_errors,
+                        old_tx_errors: tx_errors,
+                        // rx_compressed,
+                        // old_rx_compressed: rx_compressed,
+                        // tx_compressed,
+                        // old_tx_compressed: tx_compressed,
+                        updated: true,
+                    });
+                }
+            };
+        }
+
+        // Remove interfaces that are gone
+        for name in interfaces
```

Comment:
Instead of all this loop, I can now recommend you this method: https://doc.rust-lang.org/nightly/std/collections/hash_map/struct.HashMap.html#method.retain :D

Reply:
Done, I learned a new one.